### PR TITLE
Make `l1_origin` in `L2BlockRef` a struct instead of an enum

### PR DESCRIPTION
**Pull request by HrikB**
Currently, `l1_origin` in the `L2BlockRef` struct is an enum. But from the link embedded above it, it is supposed to be a struct. 

See: https://github.com/ethereum-optimism/optimism/blob/develop/op-service/eth/id.go#L10-L13

### CODE REVIEW

File: crates/rpc-types/src/sync.rs [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
 
+
+/// See: https://github.com/ethereum-optimism/optimism/blob/develop/op-service/eth/id.go#L10-L13
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct BlockID {
+    /// The block hash.
+    pub hash: B256,
+    /// The block number.
+    pub number: BlockNumber,
+}
```
Comment (mattsse) [inline crates/rpc-types/src/sync.rs:14]: this can be alloy's BlockNumHash type

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/rpc-types/src/sync.rs (hunk 1 above)
Comment (HrikB) [reply crates/rpc-types/src/sync.rs:14]: It says `BlockNumHash` doesn't implement `Deserialize`

Comment (mattsse) [reply crates/rpc-types/src/sync.rs:14]: added https://github.com/alloy-rs/alloy/pull/1277

Comment (HrikB) [reply crates/rpc-types/src/sync.rs:14]: Should I set the `rev` to that commit or wait for a release?

Comment (mattsse) [reply crates/rpc-types/src/sync.rs:14]: can put this on hold after you changed it to blockNumHash until the next release

Comment (HrikB) [reply crates/rpc-types/src/sync.rs:14]: So just leave it for now?

Comment (mattsse) [reply crates/rpc-types/src/sync.rs:14]: > can put this on hold after you changed it to blockNumHash until the next release

**Review by refcell — APPROVED** [review]

**Comment by mattsse** [comment]
@refcell this can be closed?

**Comment by refcell** [comment]
> @refcell this can be closed?

I think this is still valid actually

**Review by mattsse — APPROVED** [review]

**Pull request merged**